### DISCUSSION TITLE
Shelly Plug S / Shelly Plus 1PM - Abfrage von Leistung oder Zählerstand ?

### DISCUSSION THREAD

**Discussion by bebu259** [camueller/SmartApplianceEnabler#521] [Installation und Betrieb]
Hallo Axel,

ich habe ein **Shelly Plug S** und ein **Shelly Plus 1PM**, die beide sowohl die `Leistung` als auch den `Zählerstand` liefern können (beim Plug S in Watt-Minuten, beim Plus1 PM in Watt-Stunden).

Der Zählerstand wird allerdings nicht gespeichert, startet also wieder bei 0, wenn die Zähler vom Netz getrennt wurden (bei Abschalten der Sicherung oder Stromausfall).

Ich bin mir jetzt nicht sicher, ob es besser ist, die Leistung oder den Zählerstand abzufragen. 
Du hattest in den Docs irgendwo geschrieben, dass der Zählerstand genauer ist (und das bei weniger Abfragen).
Den `SDM-72D` habe ich auch per Zählerstand eingebunden, aber der hat ja auch einen Speicher.

Was passiert, wenn die Zähler durch einen Stromunterbrechung wieder bei 0 starten?
Wird das bei der Berechnung der Leistung (Differenz der Zählerstände) abgefangen oder würde dann eine (mitunter große) negative Leistung gemeldet?

Hier noch die Infos zu den beiden Zählern:

**Shelly Plug S** Abfrage über `http://MY_IP_ADDRESS/meter/0`:
<img width="452" alt="Bildschirmfoto 2023-05-02 um 18 48 45" src="https://user-images.githubusercontent.com/1553063/235735348-773c0b18-f155-405f-9828-32db0d4fc1f5.png">

**Shelly Plus 1PM** Abfrage über `http://MY_IP_ADDRESS/rpc/Shelly.GetStatus`:

<img width="336" alt="Bildschirmfoto 2023-05-02 um 18 59 03" src="https://user-images.githubusercontent.com/1553063/235735177-d6d999bf-4b85-4501-80a2-4c416da4cdfb.png">

**Comment by camueller** [comment]
> Was passiert, wenn die Zähler durch einen Stromunterbrechung wieder bei 0 starten?

Ist kein Problem für den SAE.

**Reply by bebu259** [reply]
👍 Top, dann frage ich die Zählerstände ab, vielen Dank!
